Il2CppInspector.Common/Reflection/Il2CppModel.cs

```diff
@@ -22,18 +22,20 @@ public class Il2CppModel
         // List of all types from TypeRefs ordered by instanceIndex
         public TypeInfo[] TypesByReferenceIndex { get; }
 
-        // List of all types from MethodSpecs (closed generic types that can be instantiated)
-        public Dictionary<int, TypeInfo> TypesByMethodSpecClassIndex { get; } = new Dictionary<int, TypeInfo>();
-
         // List of all methods from MethodSpecs (closed generic methods that can be called; does not need to be in a generic class)
         public Dictionary<Il2CppMethodSpec, MethodBase> GenericMethods { get; } = new Dictionary<Il2CppMethodSpec, MethodBase>();
 
         // List of all type definitions by fully qualified name (TypeDefs only)
         public Dictionary<string, TypeInfo> TypesByFullName { get; } = new Dictionary<string, TypeInfo>();
 
         // Every type
-        public IEnumerable<TypeInfo> Types => new IEnumerable<TypeInfo>[] {TypesByDefinitionIndex, TypesByReferenceIndex, TypesByMethodSpecClassIndex.Values}
-                .SelectMany(t => t).Distinct();
+        public IEnumerable<TypeInfo> Types {
+            get {
+                var result = new IEnumerable<TypeInfo>[] { TypesByDefinitionIndex, TypesByReferenceIndex }.SelectMany(t => t);
+                result = result.Concat(result.SelectMany(t => t.CachedGeneratedTypes));
```

Comment:
Is the cache guaranteed to be fully populated before this is called?

Reply:
`TypesByDefinitionIndex` and `TypesByReferenceIndex` will be fully populated, but the cache may not be fully populated. Note that the cache lists types that aren't directly listed in the metadata, but are referenced somehow (e.g. through members of a concrete generic type).